File: Donate.py
import History

class Donate:
    def __init__(self, history: History):
        self.history = history
        self.donate = 2000
        
    def getDonate(self) -> int:
        productionList = self.history.getProduction()
        count = 0
        
        for j in range(len(productionList) - 1):
            if productionList[j] < productionList[j + 1]:
                count += 1
                
        return self.donate * count

File: src/main/java/com/ja/model/part/Donate.java
package com.ja.model.part;

public class Donate {

    private History history;
    private final int donateValue = 2000;

    public Donate(History history) {
        this.history = history;
    }

    public int getDonate() {
        if(history.getDayMax() - 1 == history.getDayCounter()) {
            int count = 0;

            for(int i = 0; i < history.getDayMax() - 1; i++) {
                if(history.getProduction(i) < history.getProduction(i + 1)) {
                    count++;
                }
            }

            return count * donateValue;
        }

        else {
            throw new RuntimeException("Donate: Wrong Day!");
        }
    }
}
